src/eos/radiationProperties/radiationZimmer.cpp

```diff
@@ -0,0 +1,241 @@
+#include "radiationZimmer.hpp"
+#include "math.h"
+
+ablate::eos::radiationProperties::Zimmer::Zimmer(std::shared_ptr<eos::EOS> eosIn) : eos(std::move(eosIn)) {}
+
+PetscErrorCode ablate::eos::radiationProperties::Zimmer::ZimmerFunction(const PetscReal *conserved, PetscReal *kappa, void *ctx) {
+    PetscFunctionBeginUser;
+
+    /** This model depends on mass fraction, temperature, and density in order to predict the absorption properties of the medium. */
+    auto functionContext = (FunctionContext *)ctx;
+    double temperature, density;  //!< Variables to hold information gathered from the fields
+    PetscErrorCode ierr;          //!< Standard PETSc error code returned by PETSc functions
+
+    ierr = functionContext->temperatureFunction.function(conserved, &temperature, functionContext->temperatureFunction.context.get());  //!< Get the temperature value at this location
+    CHKERRQ(ierr);
+    ierr = functionContext->densityFunction.function(conserved, &density, functionContext->densityFunction.context.get());  //!< Get the density value at this location
+    CHKERRQ(ierr);
+
+    double kapparef = 1;  //!< Reference absorptivity
+    double Tsurf = 300.;  //!< Reference temperature in Kelvin
+
+    /** The Zimmer model uses a fit approximation of the absorptivity. This depends on the presence of four species which are present in combustion and shown below. */
+    double kappaH2O = 0;
+    double kappaCO2 = 0;
+    double kappaCH4 = 0;
+    double kappaCO = 0;
+    double pCO2, pH2O, pCH4, pCO;
+
+    /** These are the constants stored for the model. */
+    std::array<double, 7> H2O_coeff[] = {{0.22317e1, -.15829e1, .1329601e1, -.50707, .93334e-1, -0.83108e-2, 0.28834e-3}};       //
```

Comment:
Would you mind moving these arrays so that they don't need to be defined every time this function is called? Maybe move to the header file? I think that you may be defining an array of arrays.  Can we get rid of the []?